include/ultra64/abi.h

```diff
@@ -461,14 +482,21 @@ typedef short ENVMIX_STATE[40];
         _a->words.w1 = (u32)(addr);                                     \
 }
 
-#define aDuplicate(pkt, count, dmemi, dmemo, a4)                        \
-{                                                                       \
-        Acmd *_a = (Acmd *)pkt;                                         \
-                                                                        \
-        _a->words.w0 = (_SHIFTL(A_DUPLICATE, 24, 8) |                   \
-                        _SHIFTL(count, 16, 8) | _SHIFTL(dmemi, 0, 16)); \
-        _a->words.w1 = _SHIFTL(dmemo, 16, 16) | _SHIFTL(a4, 0, 16);     \
-}
+/*
+ * Duplicates 128 bytes of data a number of times.
+ *
+ * @param pkt pointer to a Acmd command
+ * @param numCopies number of times to duplicate 128 bytes from src
+ * @param dmemAddrSrc DMEM source address
+ * @param dmemAddrDest DMEM destination address for the first duplicate, each subsequent duplicate is at dest += 0x80
```

Comment:
```suggestion
 * @param dmemAddrDest DMEM destination address for the duplicates, size 128 * numCopies
```
Maybe?

Reply:
Yeah that's fine. While you're taking a look at this, what do you think of `dmemAddrDest` vs. `dmemDest` vs `dmemAddrOut` vs `dmemOut`

i.e.
1) Is the `Addr` needed? I think it helps explicitly with clarity, but maybe its redundant
2) `src/dest` vs. `in/out`. I've seen `in/out` used for dmem here previously, but `src/dest` seems more consistent with the repo as a whole

Reply:
I think dropping `addr` is fine since it can be understood that `dmem` carries the information that it's an address. I don't have strong opinions either way for `in/out` or `src/dest`, I suppose we should prefer the latter for consistency.